starknet_py/cairo/serialization/function_serialization_adapter.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import List, Tuple
+
+from starknet_py.cairo.felt import CairoData
+from starknet_py.cairo.serialization.data_serializers.payload_serializer import (
+    PayloadSerializer,
+)
+from starknet_py.cairo.serialization.errors import InvalidTypeException
+from starknet_py.utils.tuple_dataclass import TupleDataclass
+
+
+@dataclass
+class FunctionSerializationAdapter:
+    """
+    Class serializing *args and **kwargs  by adapting them to function inputs.
+    """
+
+    inputs_serializer: PayloadSerializer
+    outputs_deserializer: PayloadSerializer
```

Comment:
What's the usecase for having different serializers for inputs and outputs?

Reply:
`function(id: felt) -> User` has different inputs and outputs.

Reply:
I'm confused because in the test specifically you create a Adapter with inputs and outputs serializers exactly the same.

Reply:
Changed them to be different